benchexec/tablegenerator/statisticstex.py

```diff
@@ -0,0 +1,346 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2007-2020 Dirk Beyer <https://www.sosy-lab.org>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import copy
+import logging
+import re
+from collections import Counter, defaultdict
+from typing import List, Iterable, Set
+
+from benchexec.tablegenerator.columns import Column, ColumnType
+
+from benchexec.tablegenerator import util
+from benchexec.tablegenerator.statistics import ColumnStatistics, StatValue
+
+TEX_HEADER = r"""% The following definition defines a command for each value.
+% The command name is the concatenation of the first six arguments.
+% To override this definition, define \StoreBenchExecResult with \newcommand before including this file.
+% Arguments: benchmark name, runset name, column title, column, status, statistic, value
+\providecommand\StoreBenchExecResult[7]{\expandafter\newcommand\csname#1#2#3#4#5#6\endcsname{#7}}%
+"""
+
+RENAME_FUNCTIONS = {
+    "column": lambda value: "all" if value.lower() == "total" else value,
+}
+
+
+class LatexCommand:
+    """Data holder for latex command."""
+
+    def __init__(self, benchmark_name="", runset_name=""):
+        self.benchmark_name = LatexCommand.format_command_part(str(benchmark_name))
+        self.runset_name = LatexCommand.format_command_part(str(runset_name))
+        self.column_title = ""
+        self.column = ""
+        self.status = ""
+        self.stat_type = ""
+        self.value = None
+
+    def set_command_part(self, part_name: str, part_value) -> "LatexCommand":
+        """Sets the value of the command part
+
+        Available part names:
+            benchmark_name, runset_name, column_title, column, status, stat_type
+
+        Args:
+            part_name: One of the names above
+            part_value: The value to be set for this command part
+
+        Returns:
+            This LatexCommand
+        """
+        if part_name in RENAME_FUNCTIONS.keys():
+            part_value = RENAME_FUNCTIONS[part_name](part_value)
+        self.__dict__[part_name] = LatexCommand.format_command_part(str(part_value))
+        return self
+
+    def set_command_value(self, value: str) -> "LatexCommand":
+        """Sets the value for this command
+
+        The value must be formatted. No checks are made in this method.
+        It will be converted to string.
+
+        Args:
+            value: The new command value
+        Returns:
+            This LatexCommand
+        """
+        if value is None:
+            value = ""
+        self.value = str(value)
```

Comment:
This is somewhat confusing. Do we assume that the value is already a string, or not?

If yes: The body of the method can be replaced with `self.value = value`.
If not: The type signature of the method needs to be adjusted.

The doc string is also similarly confusing: If the value already needs to be formatted, why is it converted to string?